### ISSUE TITLE
New livewire 2.3.2 does not respect associative array key

### ISSUE THREAD

**Issue by l3lackheart** [livewire/livewire#2099]
### Description
Since new update of livewire v2.3.2, our code run to serveral problems because livewire keep reset our assoc array key
### Exact steps to reproduce
 * Create an associative array with custom numeric string key
 * Call it on blade
### Stripped-down, copy-pastable code snippets
 * app\Http\Livewire\Test.php
```
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class Test extends Component
{
    public $test = [
        '1' => [
            '123' => 'A',
            '13545' => 'B'
        ],
        '3' =>  'A',
        '4' => 'R',
        '0' => 'O'
    ];

    public function render()
    {
        return view('livewire.test');
    }
}
```
 * resources/views/livewire/test.blade.php
```
<div>
    @dump($test)
</div>
```
 * **Expect to see**
![image](https://user-images.githubusercontent.com/25253808/100307091-92a9ae80-2fd7-11eb-9efc-502febb9c09e.png)
  
 * **Things happen**
![image](https://user-images.githubusercontent.com/25253808/100307139-afde7d00-2fd7-11eb-88be-b5a680664bcd.png)

  
**Context**
- Livewire version: 2.3.2
- Laravel version: 8.16.1
- Alpine version: 2.7.3
- Browser: Chrome

**Comment by JacobHollis** [comment]
I'm having the same issue in multiple projects.

It has to be happening at some point during the render because the Livewire component still has the keys if you DD them for example, the keys gets reset when they are passed to the view.

**Comment by JacobHollis** [comment]
Looks like this is about to get fixed by this commit: https://github.com/livewire/livewire/commit/c6dee57e7d3902a9ed60c249da1a1c5367bb62d9

In case that helps you, @l3lackheart.

**Comment by ahmad-azizi-dev** [comment]
I have this problem after updating livewire to version 2.3.2
the keys of the array are reindexed!!! it will happen just for numeric type keys, not for string.

**Comment by l3lackheart** [comment]
> Looks like this is about to get fixed by this commit: [c6dee57](https://github.com/livewire/livewire/commit/c6dee57e7d3902a9ed60c249da1a1c5367bb62d9)
> 
> In case that helps you, @l3lackheart.

/sparkle
Yeah it seems good, bro
Looking forward to seeing it been merged

**Comment by Banquo1** [comment]
That gives me a lot of headaches in the last days. This commit fixed it for me. Looking forward for a new 2.3.3 tag for this fix.

**Comment by nuernbergerA** [comment]
It's already fixed but not released yet.

Feel free to use`dev-master` or lock a stable version in your `composer.json` to fix this issue.

**Comment by austenc** [comment]
Hey folks, please try out the latest version (`2.3.4`). A fix was released in 2.3.3 that should fix this issue, but 2.3.4 also includes a fix for entangle, so I would recommend updating to it.

**Comment by squishythejellyfish** [comment]
👋 Oh Hi! I'm Squishy, the friendly jellyfish that manages Livewire issues.

I see this issue has been closed.

Here in the Livewire repo, we have an "issues can be closed guilt-free and without explanation" policy.

If for ANY reason you think this issue hasn't been resolved, PLEASE feel empowered to re-open it.

Re-opening actually helps us track which issues are a priority.

Reply "REOPEN" to this comment and we'll happily re-open it for you!

(More info on this philosophy here: https://twitter.com/calebporzio/status/1321864801295978497)
